youtube_dl/extractor/laola1tv.py

```diff
@@ -1,86 +1,109 @@
 
         partner_id = self._search_regex(
-            r'partnerid\s*:\s*"([^"]+)"', iframe, 'partner id')
+            r'partnerid\s*:\s*"([^"]+)"', iframe, 'partner ID')
 
         xml_url = ('http://www.laola1.tv/server/hd_video.php?' +
                    'play=%s&partner=%s&portal=%s&v5ident=&lang=%s' % (
                        video_id, partner_id, portal, lang))
         hd_doc = self._download_xml(xml_url, video_id)
 
-        title = xpath_text(hd_doc, './/video/title', fatal=True)
-        flash_url = xpath_text(hd_doc, './/video/url', fatal=True)
-        uploader = xpath_text(hd_doc, './/video/meta_organistation')
-        is_live = xpath_text(hd_doc, './/video/islive') == 'true'
+        _v = lambda x, **k: xpath_text(hd_doc, './/video/' + x, **k)
+        title = _v('title', fatal=True)
 
-        categories = xpath_text(hd_doc, './/video/meta_sports')
-        if categories:
-            categories = categories.split(',')
+        categories = _v('meta_sports')
+        if categories: categories = categories.split(',')
 
-        ident = random.randint(10000000, 99999999)
-        token_url = '%s&ident=%s&klub=0&unikey=0&timestamp=%s&auth=%s' % (
-            flash_url, ident, flashvars['timestamp'], flashvars['auth'])
+        time_date = _v('time_date'); time_start = _v('time_start')
+        if time_date and time_start:
+            upload_date = unified_strdate(time_date + ' ' + time_start)
 
+        json_url = ('https://club.laola1.tv/sp/laola1/api/v3/user/session' +
+                    '/premium/player/stream-access?videoId=%s&target=2' +
+                    '&label=laola1tv&area=%s') % (video_id, _v('area'))
+        req = sanitized_Request(json_url, urlencode_postdata(
+            dict((i, v) for i, v in enumerate(_v('req_liga_abos').split(',')))))
+
+        token_url = self._download_json(req, video_id)['data']['stream-access'][0]
         token_doc = self._download_xml(
             token_url, video_id, note='Downloading token')
-        token_attrib = token_doc.find('.//token').attrib
-        if token_attrib.get('auth') in ('blocked', 'restricted'):
-            raise ExtractorError(
-                'Token error: %s' % token_attrib.get('comment'), expected=True)
 
-        video_url = '%s?hdnea=%s&hdcore=3.2.0' % (
-            token_attrib['url'], token_attrib['auth'])
+        _t = lambda x: xpath_attr(token_doc, './/token', x)
+        token_auth = _t('auth'); token_url = _t('url')
+
+        if token_auth in ('blocked', 'restricted'):
+            raise ExtractorError('Token error: %s' % _t('comment'), expected=True)
+        elif token_auth and token_url:
+            video_url = '%s?hdnea=%s&hdcore=3.2.0' % (token_url, token_auth)
 
         return {
+            'categories': categories,
+            'ext': 'mp4',
             'id': video_id,
-            'is_live': is_live,
+            'is_live': _v('islive') == 'true',
             'title': title,
+            'upload_date': upload_date,
```

Comment:
This will result in reference to unassigned variable when time fields are missing.